pkg/protocol/commitment_verifier.go

```diff
@@ -55,21 +65,37 @@ func (c *CommitmentVerifier) verifyCommitment(prevCommitment, commitment *model.
 		tree.Set(att.IssuerID, att)
 	}
 	if !iotago.VerifyProof(merkleProof, iotago.Identifier(tree.Root()), commitment.RootsID()) {
-		return nil, ierrors.Errorf("invalid merkle proof for attestations for commitment %s", commitment.ID())
+		return nil, 0, ierrors.Errorf("invalid merkle proof for attestations for commitment %s", commitment.ID())
 	}
 
 	// 2. Verify attestations.
 	blockIDs, seatCount, err := c.verifyAttestations(attestations)
 	if err != nil {
-		return nil, ierrors.Wrapf(err, "error validating attestations for commitment %s", commitment.ID())
+		return nil, 0, ierrors.Wrapf(err, "error validating attestations for commitment %s", commitment.ID())
 	}
 
-	// 3. Verify cumulative weight of commitment matches with calculated weight from attestations.
-	if prevCommitment.CumulativeWeight()+seatCount != commitment.CumulativeWeight() {
-		return nil, ierrors.Errorf("invalid cumulative weight for commitment %s", commitment.ID())
+	// 3. Verify that cumulative weight of commitment matches (as an upper bound) with calculated weight from attestations.
+	// This is necessary due to the following edge case that can happen in the window of forking point and the current state of the other chain:
+	//	1. A public key is added to an account.
+	//     We do not count a seat for the issuer for this slot and the computed CW will be lower than the CW in
+	//	   the commitment. This is fine, since this is a rare occasion and a heavier chain will become heavier anyway, eventually.
+	//	   It will simply take a bit longer to accumulate enough CW so that the chain-switch rule kicks in.
+	//	TODO: what happens in an extreme case where all change their keys?
```

Comment:
I don't believe this situation can be solved: how can you verify data on-chain if you only trust the forking point, in which, all the keys are now discontinued?
You cannot verify the chain up-front: your only chance is to actually sync the chain, regardless of its CW and, once synced, check its candidate CW against your main CW.

Reply:
I added a note that the only solution to prevent this is to add changes (provable) to the commitments